src/dfx/src/commands/ledger/fabricate_cycles.rs

```diff
@@ -0,0 +1,104 @@
+use crate::lib::environment::Environment;
+use crate::lib::error::DfxResult;
+use crate::lib::identity::identity_utils::CallSender;
+use crate::lib::models::canister_id_store::CanisterIdStore;
+use crate::lib::operations::canister;
+use crate::lib::root_key::fetch_root_key_or_anyhow;
+use crate::util::clap::validators::{cycle_amount_validator, trillion_cycle_amount_validator};
+use crate::util::expiry_duration;
+
+use clap::Parser;
+use ic_types::Principal;
+use slog::info;
+use std::time::Duration;
+
+const DEFAULT_CYCLES_TO_FABRICATE: u128 = 10_000_000_000_000_u128;
+
+/// Local development only: Fabricate cycles out of thin air and deposit them into the specified canister(s).
+#[derive(Parser)]
+pub struct FabricateCyclesOpts {
+    /// Specifies the amount of cycles to fabricate. Defaults to 10T cycles.
+    #[clap(long, validator(cycle_amount_validator), conflicts_with("t"))]
+    amount: Option<String>,
+
+    /// Specifies the amount of trillion cycles to fabricate. Defaults to 10T cycles.
+    #[clap(
+        long,
+        validator(trillion_cycle_amount_validator),
+        conflicts_with("amount")
+    )]
+    t: Option<String>,
```

Comment:
Can we use `icp` and `e8s` arguments instead of `t` here, to be consistent with `dfx ledger top-up`, which is the equivalent mainnet command?

https://github.com/dfinity/sdk/blob/master/src/dfx/src/commands/ledger/top_up.rs#L31

Reply:
I do not think this is possible this easily. `provisional_top_up_canister` expects an amount in cycles, but `--e8s/--icp` would be an amount in ICP. We don't have an exchange rate set, therefore there's no ICP price to convert to cycles.

I'm trying to figure out how we can query the exchange rate, but it's not looking great

Reply:
Ah of course, you're right.  Nevermind!